src/esoh/ingest/main.py

```diff
@@ -68,10 +74,13 @@ def _decide_input_type(self, message) -> str:
         """
         Internal method for deciding what type of input is being provided.
         """
-        match message.split(".")[-1]:
+        file_name = os.path.basename(message)
+        if re.match("data[0-9][0-9][0-9][0-9]", file_name):
```

Comment:
Why are this regex expression needed? What file ending is this handeling?

Reply:
For example data0100 . This is real GTS data.

Reply:
Ok, I think we should also verify that there is no file ending, incase we get some data with such a name. Is this for all GTS data, or just norcom?
```suggestion
        if re.match(r"data[0-9][0-9][0-9][0-9]$", file_name):
```
Here I have added regex to require end of string. This is to ensure to not capture any other files by mistake.
When using regex it is norm to make the regex expression a raw string, ensuring that nothing is interpreted before the expression is passed to regex. This is done by putting an r in front of the string.

Reply:
These files is created by norcom. I can't access to GTS directly. The norcomm puts the BUFR messages(with GTS header) to these files from GTS. The decoder can handle GTS data directly.